### PR TITLE
[bug] Fixed dec and dsc parser to explode less

### PR DESCRIPTION
**Pull request by matthewfcarlson**
Previously, when the dec or dsc parser encounters an error, it explodes. Previously the API didn't really expect callers to handle exceptions.

### CODE REVIEW

File: edk2toollib/uefi/edk2/parsers/dec_parser.py [hunk 1]
```diff
@@ -199,6 +200,17 @@ def _Parse(self) -> None:
             elif InPcdSection:
                 if sline.strip()[0] == '[':
                     InPcdSection = False
+                elif sline.strip().endswith("{"):
```
Comment (corthon) [inline edk2toollib/uefi/edk2/parsers/dec_parser.py]: Is it sufficient to only check the end of the line? Are '{' in the middle of a line a concern? Or has it been preprocessed by this point in the code?

**Comment by matthewfcarlson** [comment]
This will need to be merged in with #71

File: edk2toollib/uefi/edk2/parsers/dec_parser.py (hunk 1 above)
Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dec_parser.py]: It will be pre-processed, so comments will be stripped and blank lines skipped

Comment (corthon) [reply edk2toollib/uefi/edk2/parsers/dec_parser.py]: There's no possibility of an inline struct or something? Like...
```
PcdSomethingSomething|{blah, blah}
```

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dec_parser.py]: That's a great point.

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dec_parser.py]: Maybe just look at if the count of({) > count of (})

Comment (corthon) [reply edk2toollib/uefi/edk2/parsers/dec_parser.py]: Dunno. Just trying to poke holes in your work. :-p

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dec_parser.py]: No it's good :)

**Comment by matthewfcarlson** [comment]
Is this still needed after #73?

**Pull request closed without merging**